angularSTEP/src/app/signup/signup.component.spec.ts

```diff
@@ -0,0 +1,269 @@
+import { async, ComponentFixture, TestBed} from '@angular/core/testing';
+import {MatButtonHarness} from '@angular/material/button/testing';
+import {HarnessLoader} from '@angular/cdk/testing';
+import {TestbedHarnessEnvironment} from '@angular/cdk/testing/testbed';
+import {SignupComponent} from './signup.component';
+import {BrowserModule, By} from '@angular/platform-browser';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import {MatTabsModule} from '@angular/material/tabs'
+import {AngularFireModule} from '@angular/fire';
+import {environment} from '../../environments/environment';
+import {AngularFirestoreModule, AngularFirestore} from '@angular/fire/firestore';
+import {Router} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {MatFormFieldModule} from '@angular/material/form-field';
+import {MatInputModule} from '@angular/material/input';
+import {MatButtonModule} from '@angular/material/button';
+import {MatIconModule} from '@angular/material/icon';
+import {AngularFireAuthModule, AngularFireAuth} from '@angular/fire/auth';
+import {FormBuilder} from '@angular/forms';
+
+
+const docExists = {exists: true};
+const docDoesntExist = {exists: false};
+const AngularFirestoreStub = {
+  doc: jasmine.createSpy('doc').withArgs("/usernames/exists/").and.returnValue({
```

Comment:
nit: It's nice to consistently use either " or ' for string quotes as much as possible within a file